### PR TITLE
Navbar fixes

### PR DESCRIPTION
**Pull request by AKeeganDev**
LOTS of BUZZ with this pull request! The following components/pages have been modified:

- Navbar
  - Fixed bugs where the hamburger or nav menu would lay over other popups 
  - Now closes when a user selects a navlink OR clicks outside the navbar
  - Now displays a sign up link below the login. Both login and Sign up can only be seen when the user is not logged in
  - When a user is logged in then the login and signup buttons are hidden and replaced with a functional Sign Out link

- Login Page
  - A user can now sign in if they use a valid account name from the database.
  - Once signed in they are directed back to the page they were originally viewing
 
- Sign Up Page
  - Users can now sign up for a new account and the account name is saved to the database
  - When a user account creation is successful then they are displayed a message welcoming them to the site and they are automatically signed in

### CODE REVIEW

File: src/Components/Navbar.js [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
 
 const Navbar = () => {
+  const user = useSelector((state) => state.user);
+  const dispatch = useDispatch();
+
+  const handleClick = () => {
+    dispatch(logOutUser());
+  };
+
+  console.log(handleClick);
```
Comment (hsztan) [inline src/Components/Navbar.js:26]: - Is this console output necessary?

File: src/Components/Navbar.js [hunk 2]
```diff
@@ -1,14 +1,49 @@
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { NavLink } from 'react-router-dom';
 import { toggleLogin } from '../Pages/LoginPage';
+import { toggleSignup } from '../Pages/SignupPage';
+import { logOutUser } from '../Redux/user/UserReducer';
+// import { toggleSignup } from '../Pages/SignupPage';
```
Comment (hsztan) [inline src/Components/Navbar.js:7]: - Commented is better removed than left inline.

**Review by hsztan — APPROVED** [review]
-- Very detailed PR.
-- Code is readable, with good variable names.
-- Great work on the state!

giphy

File: src/Components/Navbar.js (hunk 1 above)
Comment (AKeeganDev) [reply src/Components/Navbar.js:26]: Oops! Nope! I'll comb through the consoles and take them out. Totally spaced on that. Good catch

File: src/Components/Navbar.js (hunk 2 above)
Comment (AKeeganDev) [reply src/Components/Navbar.js:7]: @hsztan Done. good catches.

**Pull request merged**